src/os/portable/os-impl-posix-dl.c

```diff
@@ -51,6 +51,7 @@
  */
 typedef struct
 {
+    /* cppcheck-suppress unusedStructMember */
```

Comment:
This seems suspicious because the dl_handle is absolutely used, but the code is conditional upon `OS_MAX_MODULES > 0` in the config.

Maybe cppcheck is complaining because it doesn't know what OS_MAX_MODULES is set to, therefore assumes that the conditional is false?

Reply:
Right.  Cppcheck is checking both conditions.  I prefer this specific suppression vs passing in OS_MAX_MODULES to cppcheck, in that I don't really care about this specific warning but if future ones show up for that condition being false I want to know.  

Alternatively, if the condition really can never happen, then it probably shouldn't be there (and add a verify that OS_MAX_MODULES > 0).

Reply:
The condition of OS_MAX_MODULES being 0 _can_ happen.  But in that case the structure isn't instantiated at all. cppcheck just does not know that.

I think a better long-term solution would be to work this into the cmake-based osconfig options (to replace the preprocessor-based logic).  This will get rid of these types of `#if` conditionals in here and probably clear up this warning too.